crates/sui-core/src/unit_tests/transaction_deny_tests.rs

```diff
@@ -0,0 +1,238 @@
+    let obj_ref = accounts[0].2[0];
+    let state = TestAuthorityBuilder::new()
+        .with_transaction_deny_config(
+            TransactionDenyConfigBuilder::new()
+                .add_denied_object(obj_ref.0)
+                .build(),
+        )
+        .build_with_store(
+            network_config.genesis.committee().unwrap(),
+            network_config.validator_configs[0].protocol_key_pair(),
+            state.db(),
+            &[],
+        )
+        .await;
+    assert_denied(&transfer_with_account(&accounts[0], &state).await);
+}
+
+#[tokio::test]
+async fn test_address_denied() {
+    // We need to create the authority state once to get one of the account addresses.
+    let (network_config, state) = setup_test(TransactionDenyConfigBuilder::new().build()).await;
+    let accounts = get_accounts_and_coins(&network_config, &state);
+    // Re-create the state such that we could specify an address to be denied.
+    let state = TestAuthorityBuilder::new()
+        .with_transaction_deny_config(
+            TransactionDenyConfigBuilder::new()
+                .add_denied_address(accounts[0].0)
+                .build(),
+        )
+        .build_with_store(
+            network_config.genesis.committee().unwrap(),
+            network_config.validator_configs[0].protocol_key_pair(),
+            state.db(),
+            &[],
+        )
+        .await;
+    assert_denied(&transfer_with_account(&accounts[0], &state).await);
+}
+
+#[tokio::test]
+async fn test_shared_object_transaction_disabled() {
+    let (network_config, state) = setup_test(
+        TransactionDenyConfigBuilder::new()
+            .disable_shared_object_transaction()
+            .build(),
+    )
+    .await;
+    let accounts = get_accounts_and_coins(&network_config, &state);
+    let gas_price = state.reference_gas_price_for_testing().unwrap();
+    let tx = make_staking_transaction(
+        accounts[0].2[0],
+        accounts[0].2[1],
+        SuiAddress::default(),
+        accounts[0].0,
+        &accounts[0].1,
+        gas_price,
+    );
+    let result = state
+        .handle_transaction(&state.epoch_store_for_testing(), tx)
+        .await;
+    assert_denied(&result);
+}
+
+#[tokio::test]
+async fn test_package_publish_disabled() {
+    let (network_config, state) = setup_test(
+        TransactionDenyConfigBuilder::new()
+            .disable_package_publish()
+            .build(),
+    )
+    .await;
+    let accounts = get_accounts_and_coins(&network_config, &state);
+    let rgp = state.reference_gas_price_for_testing().unwrap();
+    let mut path = PathBuf::from(env!("CARGO_MANIFEST_DIR"));
+    path.push("src/unit_tests/data/object_basics");
+    let tx = make_publish_package(accounts[0].2[0], path, rgp);
+    let result = state
+        .handle_transaction(&state.epoch_store_for_testing(), tx)
+        .await;
+    assert_denied(&result);
+}
+
+#[tokio::test]
+async fn test_package_denied() {
```

Comment:
Probably worth adding a test where you deny access to an upgraded package but not the original version?  That's the path that would be affected by the linkage table change I mentioned earlier.